scripts/profiler/processed_events.py

```diff
@@ -27,37 +32,127 @@ def __init__(self):
-                    if self.raw_data.events[j].data[0] == self.start_event.data[0] and \
-                      self.raw_data.registered_events_types[self.raw_data.events[j].type_id].data_descriptions[0] == MEM_ADDRESS_DATA_DESC:
-                        self.submit_event = self.raw_data.events[j]
-                        break
-            # comparing memory addresses of event processing start and end
-            # to identify matching events
-            if self.raw_data.events[i].type_id == self.event_processing_end_id:
-                if self.submit_event is not None \
-                        and self.raw_data.events[i].data[0] == self.start_event.data[0]:
-                    self.tracked_events.append(
-                        TrackedEvent(
-                            self.submit_event,
-                            self.start_event.timestamp,
-                            self.raw_data.events[i].timestamp))
-                    self.submit_event = None
+    def get_event_type_id(self, type_name):
+        for key, value in self.registered_events_types.items():
+            if type_name == value.name:
+                return key
+        return None
+
+    def verify(self):
+        for ev in self.tracked_events:
+            if ev.submit.type_id not in self.registered_events_types:
+                return False
+        return True
+
+    def init_writing_data_to_files(self, filename_events, filename_event_types):
+        csvfile = self._init_writing_events_csv(filename_events)
+        self._init_writing_events_types_json(filename_event_types)
+        return csvfile
+
+    def finish_writing_data_to_files(self, csvfile, filename_events, filename_event_types):
+        csvfile.close()
+        csv_hash = ProcessedEvents._calculate_md5_hash_of_file(filename_events)
+        self._write_events_types_json(filename_event_types, csv_hash)
+
+    def write_data_to_files(self, filename_events, filename_event_types):
+        self._write_events_csv(filename_events)
+        csv_hash = ProcessedEvents._calculate_md5_hash_of_file(filename_events)
+        self._write_events_types_json(filename_event_types, csv_hash)
+
+    def read_data_from_files(self, filename_events, filename_event_types):
+        csv_hash1 = self._read_events_types_json(filename_event_types)
+        self._read_events_csv(filename_events)
+        csv_hash2 = ProcessedEvents._calculate_md5_hash_of_file(filename_events)
+        if csv_hash1 != csv_hash2:
+            self.logger.warning("Hash values of csv files do not match")
+            self.logger.warning("Events and descriptions may be inconsistent")
+    @staticmethod
+    def _calculate_md5_hash_of_file(filename):
+        return hashlib.md5(open(filename, 'rb').read()).hexdigest()
+
+    def _init_writing_events_csv(self, filename):
+        try:
+            csvfile = open(filename, 'w', newline='')
+            fieldnames = ['type_id', 'timestamp', 'data', 'proc_start_time', 'proc_end_time']
```

Comment:
I would place fieldnames only in Event class and provide getter to receive it here (to keep it in one place)